core/node/rpc/sync/client/syncer_set.go

```diff
@@ -617,62 +618,66 @@ func (ss *SyncerSet) selectNodeForStream(ctx context.Context, cookie *SyncCookie
 
 // getOrCreateSyncer returns the syncer for the given node address.
 // If the syncer does not exist, it creates a new one and starts it.
-func (ss *SyncerSet) getOrCreateSyncer(nodeAddress common.Address) (StreamsSyncer, error) {
-	ss.muSyncers.Lock()
-	defer ss.muSyncers.Unlock()
-
-	if syncer, found := ss.syncers[nodeAddress]; found {
-		return syncer, nil
-	}
-
-	var syncer StreamsSyncer
-
-	if nodeAddress == ss.localNodeAddress {
-		syncer = newLocalSyncer(
-			ss.globalCtx,
-			ss.localNodeAddress,
-			ss.streamCache,
-			ss.messageDistributor,
-			ss.streamID2Syncer.Delete,
-			ss.otelTracer,
-		)
-	} else {
-		client, err := ss.nodeRegistry.GetStreamServiceClientForAddress(nodeAddress)
-		if err != nil {
-			return nil, AsRiverError(err).Tag("remoteSyncerAddr", nodeAddress)
-		}
+func (ss *SyncerSet) getOrCreateSyncer(nodeAddress common.Address) (resultSyncer StreamsSyncer, resultErr error) {
+	ss.syncers.Compute(
```

Comment:
`Compute` locks an entire hash bucket. So this approach doesn't guarantee parallel creation/retrieval of syncers.